### ISSUE TITLE
Asynchronous client

### ISSUE THREAD

**Issue by FireMasterK** [iggy-rs/iggy-python-client#8]
Currently, the client is synchronous as seen in https://github.com/iggy-rs/iggy-python-client/blob/main/src/client.rs

Are there any plans for an asynchronous client, for use with libraries such as asyncio?

**Comment by bartoszkobylinski** [comment]
Just a heads-up, if you check "python_examples" file, you'll notice we're already using asyncio. It fits like a glove since irrys is all about that async life naturally. Thought you'd like to know!

**Comment by FireMasterK** [comment]
True, I noticed that earlier too, but the actual functions aren't defined as `async`, if I understand correctly the thread from python is blocked on the tokio runtime task.

**Comment by hubcio** [comment]
> True, I noticed that earlier too, but the actual functions aren't defined as `async`, if I understand correctly the thread from python is blocked on the tokio runtime task.

hello! i'm not python expert - could you please propose PR how to fix this?

**Comment by hubcio** [comment]
@FireMasterK I asked chatgpt about this:
https://chat.openai.com/share/b102601e-6673-4315-9a95-cdae8ef4ed86

does it sounds reasonable to you?

**Comment by darioalessandro** [comment]
It is done 😄 https://github.com/iggy-rs/iggy-python-client/pull/36
